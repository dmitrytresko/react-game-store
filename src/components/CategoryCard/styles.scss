@import '/src/styles/App.scss';

.category-card {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  min-width: 90px;
  padding: 20px 0 10px;
  background: black;
  border-top: $primary-border #f8f8f8;
  border-bottom: $primary-border #f8f8f8;
  cursor: pointer;
  transition: all 0.7s ease;

  &__logo {
    width: 3.5vw;
    min-width: 35px;
    height: 3.5vw;
    min-height: 35px;
    margin-bottom: 10px;
    transition: all 0.4s ease;

    &--light {
      display: none;
      width: 3.5vw;
      min-width: 35px;
      height: 3.5vw;
      min-height: 35px;
      margin-bottom: 10px;
    }
  }

  &__name {
    color: #f8f8f8;
    font-weight: normal;
    font-size: 25px;
    font-family: $primary-font;
    text-transform: uppercase;
    cursor: pointer;
    transition: all 0.4s;

    $largeMediaWidth: 1201px;
    $maxMediaWidth: 1024px;
    $tabletMediaWidth: 768px;
    $correctedTabletMediaWidth: 767px;
    $minMediaWidth: 320px;

    @media (max-width: $largeMediaWidth) {
      font-size: 23px;
    }

    @media (max-width: $correctedTabletMediaWidth) {
      font-size: 21px;
    }

    @media (max-width: $minMediaWidth) {
      font-size: 18px;
    }
  }

  &:first-child {
    background: black;
    box-shadow: inset 0px 10px 10px -10px #f8f8f8;

    &:hover {
      border-top-color: #003791;
      border-bottom-color: #003791;

      -moz-box-shadow: inset 0 10px 10px -10px #003791,
        inset -10px 0 10px -10px #003791, inset 0 -10px 10px -10px #003791;
      -webkit-box-shadow: inset 0 10px 10px -10px #003791,
        inset -10px 0 10px -10px #003791, inset 0 -10px 10px -10px #003791;
      box-shadow: inset 0 10px 10px -10px #003791,
        inset -10px 0 10px -10px #003791, inset 0 -10px 10px -10px #003791;

      .category-card__logo {
        display: none;

        &--light {
          display: block;
        }
      }

      .category-card__name {
        color: #003791;
      }
    }
  }

  &:nth-child(2) {
    background: black;
    box-shadow: inset 0px 10px 10px -10px #f8f8f8;

    &:hover {
      border-top-color: #107c10;
      border-bottom-color: #107c10;

      -moz-box-shadow: inset 0 0px 10px #107c10;
      -webkit-box-shadow: inset 0 0px 10px #107c10;
      box-shadow: inset 0 0px 10px #107c10;

      .category-card__logo {
        display: none;

        &--light {
          display: block;
        }
      }

      .category-card__name {
        color: #107c10;
      }
    }
  }

  &:last-child {
    background: black;
    box-shadow: inset 0px 10px 10px -10px #f8f8f8;

    &:hover {
      border-top-color: #00adef;
      border-bottom-color: #00adef;

      -moz-box-shadow: inset 0 10px 10px -10px #00adef,
        inset 10px 0 10px -10px #00adef, inset 0 -10px 10px -10px #00adef;
      -webkit-box-shadow: inset 0 10px 10px -10px #00adef,
        inset 10px 0 10px -10px #00adef, inset 0 -10px 10px -10px #00adef;
      box-shadow: inset 0 10px 10px -10px #00adef,
        inset 10px 0 10px -10px #00adef, inset 0 -10px 10px -10px #00adef;

      .category-card__logo {
        display: none;

        &--light {
          display: block;
        }
      }

      .category-card__name {
        color: #00adef;
      }
    }
  }

  &:hover {
    background: rgba(255, 255, 255, 0.9);
  }
}
