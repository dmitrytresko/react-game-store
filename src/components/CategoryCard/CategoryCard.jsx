import { useHistory } from "react-router-dom";
import "./styles.scss";

const CategoryCard = (props) => {
  const history = useHistory();

  return (
    <div className="category-card" onClick={() => history.push(`${props.route}`)}>
      <img className="category-card__logo" src={props.path} alt={props.altName} />
      <h2 className="category-card__name">{props.name}</h2>
    </div>
  )
}

export default CategoryCard;
