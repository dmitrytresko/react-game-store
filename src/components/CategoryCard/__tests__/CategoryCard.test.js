import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router, Route } from "react-router-dom";
import CategoryCard from "../CategoryCard";
import playStationColorfulLogo from "../../assets/img/playstation-colorful.jpg";
import playStationLightLogo from "../../assets/img/playstation-light.jpg";

describe("Category card", () => {
  let history;
  let mockProps;

  beforeEach(() => {
    history = createMemoryHistory();

    mockProps = {
      path: playStationColorfulLogo,
      pathLight: playStationLightLogo,
      altName: "Colorful Logo",
      altNameLight: "Light Logo",
      name: "Platform",
      route: "/products/ps"
    }
  })

  it("should render Category card component", () => {
    const { container, getAllByRole } = render(
      <Router history={history}>
        <Route path="/">
          <CategoryCard {...mockProps} />
        </Route>
      </Router>
    );

    const images = getAllByRole('img');

    expect(images.length).toEqual(2);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  })

  it("should redirect to corresponding Product page", () => {
    const { container } = render(
      <Router history={history}>
        <Route path="/">
          <CategoryCard {...mockProps} />
        </Route>
      </Router>
    );

    const card = container.getElementsByClassName('category-card')[0];

    fireEvent.click(card);
    expect(history.location.pathname).toEqual(mockProps.route);
  })
})

