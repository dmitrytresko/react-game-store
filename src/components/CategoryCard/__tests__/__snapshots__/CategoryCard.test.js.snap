// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category card should render Category card component 1`] = `
<div>
  <div
    class="category-card"
  >
    <img
      alt="Colorful Logo"
      class="category-card__logo"
      src="test-file-stub"
    />
    <img
      alt="Light Logo"
      class="category-card__logo--light"
      src="test-file-stub"
    />
    <h2
      class="category-card__name"
    >
      Platform
    </h2>
  </div>
</div>
`;
